website/content/en/docs/building-operators/golang/advanced-topics.md

```diff
@@ -285,6 +285,70 @@ func main() {
+
+For convenience tools like [buildah](https://github.com/containers/buildah/blob/main/docs/buildah-bud.md#building-an-multi-architecture-image-using-a---manifest-option-requires-emulation-software) allow to cross-build and manifest multi-arch containers on one host. For instance:
+
+```shell
+for a in amd64 arm64 ppc64le s390x; do \
+  buildah bud --manifest registry/username/repo:v1 --arch $a; \
+done
+```
+
+This creates the manifest list, builds each image and adds them to the manifest list.
+
+The result can then be pushed to the desired registry.
+
+```shell
+buildah push registry/username/repo:v1
+```
+
+Docker with [buildx](https://docs.docker.com/buildx/working-with-buildx/#build-multi-platform-images) provides similar capabilities.
+
+```shell
+docker buildx build --push --platform linux/amd64,linux/arm64,linux/ppc64le,linux/s390x --tag registry/username/repo:v1 .
+```
+
+See [docker documentation](https://docs.docker.com/buildx/working-with-buildx/#build-multi-platform-images) for additional options.
+ 
+Note: the Dockerfile generated by the SDK for the operator explicitely references GOARCH=amd64 for go build. This can be amended to GOARCH=$TARGETPLATFORM. Docker will automatically set the environment variable to the value specified by --platform. With buildah --build-arg will need to be used for the purpose.
+
+All image references in the ClusterServiceVersion should be manifest lists containing the pointers to the image manifests for the supported architectures.
+
+#### Architecture specific content
+
+Operand images that require different image content (such as unique runtime packages) per architecture should be handled by the operator itself based on environment variables. That is, the CSV can list environment variables for each supported architecture / OS, for example:
+
+```
+MY_OPERAND_ARM64: my/arm64/image
+MY_OPERAND_PPC64LE: my/ppc64le/image
+```
+
+Based on the architecture and OS of the container the operator is running in, the operator can decide which environment variable to use to deploy the correct operands. When using a go-based operator the GOARCH environment variable in the container can provide the required information.
+
+#### Bundle image
+
+The bundle image only gets built for x86_64 and, since it does not contain any RPM content, it will be able to run on any architecture.
+
+#### ClusterServiceVersion labels
+
+Since all indexes for all architectures share the same content, installable operators are filtered in the UI based on the arch labels in the CSV of the head of the default channel.
+
+Labels for OS and architectures can be set in the CSV with:
+
+```
+labels:
+    operatorframework.io/arch.<GOARCH>: supported
+    operatorframework.io/os.<GOOS>: supported
```

Comment:
How does this work?

If I have an operator that supports "windows/amd64" and "linux/arm64" only, what labels would I set? It seems like I would set:

```yaml
labels:
    operatorframework.io/arch.amd64: supported
    operatorframework.io/os.windows: supported
    operatorframework.io/arch.arm64: supported
    operatorframework.io/os.linux: supported
```

But that makes it look like all combinations of arch/os are supported (i.e. `windows/amd64`, `windows/arm64`, `linux/amd64`, and `linux/arm64`).

Reply:
You are inferring a coupling between architecture and OS that may not have been foreseen in OLM. To be honest I am not sure that someone would do that: supporting one architecture with an OS and another architecture with another OS but feel free to open an OLM issue: https://olm.operatorframework.io/docs/advanced-tasks/ship-operator-supporting-multiarch/#multiple-architectures

Reply:
Yeah it seems like we really need OLM to deprecate these labels and replace them with something like this:
```yaml
labels:
    operatorframework.io/platform.linux/amd64: supported
    operatorframework.io/platform.windows/arm64: supported
```

And if we need to do that, I'm not sure it makes sense to document multi-arch support in the SDK docs until this is resolved.

Reply:
/hold

Reply:
There is a need today for providing documentation on how things are working right now. When/if labeling gets improved the documentation can reflect the new state of the world. This is just business as usual: code/api changes need to come together with the matching documentation updates.
I don't see why documentation that is already written (I mean this PR) needs to be set on hold for something that does not exist, needs to be implemented with the old version needing to go through a deprecation process.

Reply:
For the sake of progressing this PR and getting instructions for multi-arch builds in the documentation I have removed the label part of the PR and replaced it with a link to the OLM documentation. I believe it is nicer for the reader not to have to jump between pages and to get everything in one place but it is still better than having nothing.
For the records OLM does not do anything today with the architecture and os labels. It is only used by the OpenShift console for filtering.